spec/classes/openldap_server_spec.rb

```diff
@@ -463,6 +463,77 @@
           end
         end
 
+        context 'with memberof enabled', :compile do
+          let(:params) do
+            super().merge(
+              {
+                :memberof     => true,
+              }
+            )
+          end
+
+          it_behaves_like "openldap::server on #{facts[:osfamily]}"
+
+          it { should contain_openldap('cn=config') }
+          it { should contain_openldap('olcDatabase={-1}frontend,cn=config').with_attributes(
+            {
+              'objectClass'  => [
+                'olcDatabaseConfig',
+                'olcFrontendConfig',
+              ],
+              'olcDatabase'  => ['{-1}frontend'],
+            }
+          ) }
+          it { should contain_openldap('olcDatabase={2}hdb,cn=config').with_attributes(
+            {
+              'objectClass'    => [
+                'olcDatabaseConfig',
+                'olcHdbConfig',
+              ],
+              'olcAccess'      => ['{0}to * by dn.base="gidNumber=0+uidNumber=0,cn=peercred,cn=external,cn=auth" manage'],
+              'olcDatabase'    => ['{2}hdb'],
+              'olcDbDirectory' => ['/var/lib/ldap/data'],
+              'olcRootDN'      => ['cn=Manager,dc=example,dc=com'],
+              'olcRootPW'      => ['secret'],
+              'olcSuffix'      => ['dc=example,dc=com'],
+            }
+          ) }
+          it { should contain_openldap('olcOverlay={0}memberof,olcDatabase={2}hdb,cn=config').with_attributes(
+            {
+              'objectClass'  => [
+                'olcOverlayConfig',
+              ],
+              'olcOverlay'   => ['{0}memberof'],
+            }
+          ) }
+
+          case facts[:osfamily]
+          when 'Debian'
+            it { should contain_openldap('cn=module{0},cn=config').with_attributes(
+              {
+                'cn'            => ['module{0}'],
+                'objectClass'   => ['olcModuleList'],
+                'olcModuleLoad' => [
+                  '{0}back_monitor.la',
+                  '{1}back_hdb.la',
+                  '{2}unique.la',
+                ],
+              }
+            ) }
+          when 'RedHat'
+            it { should contain_openldap('cn=module{0},cn=config').with_attributes(
+              {
+                'cn'            => ['module{0}'],
+                'objectClass'   => ['olcModuleList'],
+                'olcModuleLoad' => [
+                  '{0}unique.la',
```

Comment:
Hmm... odd. I just cut and pasted one of the other checks and changed the variable name. I'll check into it and fix in the next few hours.